torch_xla/csrc/tensor.cpp

```diff
@@ -622,6 +632,10 @@ void XLATensor::SetIrValue(ir::Value ir_value) {
     AssignIrValue(std::move(ir_value));
     TryLimitGraphSize();
   }
+  if (ExecutionContext::Get()->is_eager_execution() && !copy_from_aten_tensor) {
```

Comment:
a bit confuse about `copy_from_aten_tensor`, why not just check `data()->view` if the intention is not to sync the view tensor?

Reply:
The reason for the `copy_from_aten_tensor` was to avoid a sync that happens when a copy happens via this [path](https://github.com/pytorch/xla/blob/master/torch_xla/csrc/aten_xla_type.cpp#L414) . That is when an aten tensor is copied to xla tensor. That is not an aten op, but just a copy and hence avoiding a sync here.

Reply:
I think there are two cases here
1. the copied IR is a `device_data`, this is most likely the case since `eager_debug_mode` is on. Every IR will be materized to a `device_data`(pretty much just means a data on the device side). In this case we can just check the IR type
2. The copied IR is not a `device_data`. It is very unlikely but in this case we actually want to execute this ir.

I think it is better to check if the `IR` is a `device_data` instead of `copy_from_aten_tensor`.

Device Data reference https://github.com/pytorch/xla/blob/master/torch_xla/csrc/ops/device_data.h

Reply:
Makes sense. Will make the change